src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactory.java

```diff
@@ -0,0 +1,76 @@
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.exasol.adapter.dynamodb.documentpath.DocumentPathExpression;
+
+/**
+ * This class build {@link DynamodbTableMetadata} by fetching the required information using a {@code describeTable}
+ * call to DynamoDB.
+ */
+public class DynamodbTableMetadataFactory {
+
+    /**
+     * Builds {@link DynamodbTableMetadata} for a DynamoDB table.
+     * 
+     * @param connection DynamoDB connection used for {@code describeTable} call
+     * @param tableName  DynamoDB table name
+     * @return {@link DynamodbTableMetadata} describing the keys and indexes of the table
+     */
+    public DynamodbTableMetadata forDynamodbTable(final AmazonDynamoDB connection, final String tableName) {
+        final TableDescription tableDescription = connection.describeTable(tableName).getTable();
+        final DynamodbKey primaryKey = extractKey(tableDescription.getKeySchema());
+        final List<DynamodbKey> localIndexes = extractLocalSecondaryIndex(tableDescription);
+        final List<DynamodbKey> globalIndexes = extractGlobalSecondaryIndex(tableDescription);
+        return new DynamodbTableMetadata(primaryKey, localIndexes, globalIndexes);
+    }
+
+    private DynamodbKey extractKey(final List<KeySchemaElement> keySchema) {
+        final DocumentPathExpression partitionKey = extractPartitionKey(keySchema);
+        final DocumentPathExpression sortKey = extractSortKey(keySchema);
+        return new DynamodbKey(partitionKey, sortKey);
+    }
+
+    private DocumentPathExpression extractPartitionKey(final List<KeySchemaElement> keySchema) {
+        for (final KeySchemaElement keySchemaElement : keySchema) {
+            if (keySchemaElement.getKeyType().equals("HASH")) {
+                final String keyName = keySchemaElement.getAttributeName();
+                return new DocumentPathExpression.Builder().addObjectLookup(keyName).build();
+            }
+        }
+        throw new IllegalStateException("Could not find partition key. "
+                + "That's strange because each Dynamodb table must define a partition key.");
+    }
+
+    private DocumentPathExpression extractSortKey(final List<KeySchemaElement> keySchema) {
+        for (final KeySchemaElement keySchemaElement : keySchema) {
+            if (keySchemaElement.getKeyType().equals("RANGE")) {
+                final String keyName = keySchemaElement.getAttributeName();
+                return new DocumentPathExpression.Builder().addObjectLookup(keyName).build();
+            }
+        }
+        return null;
+    }
+
+    private List<DynamodbKey> extractLocalSecondaryIndex(final TableDescription tableDescription) {
+        if (tableDescription.getLocalSecondaryIndexes() != null) {
+            return tableDescription.getLocalSecondaryIndexes().stream().map(index -> extractKey(index.getKeySchema()))
+                    .collect(Collectors.toList());
+        } else {
+            return List.of();
```

Comment:
`Collections.emptyList();` doesn't create a new object

Reply:
```
static <E> List<E> of() {
        return ImmutableCollections.emptyList();
    }

static <E> List<E> emptyList() {
        return ImmutableCollections.ListN.EMPTY_LIST;
    }
```

its the same ;-) I also discussed this with sebastian but we agreed `Collections.emptyList();` is better readable

Reply:
But it's not immutable...